edfi-repo-auditor/edfi_repo_auditor/auditor.py

```diff
@@ -0,0 +1,144 @@
+
+    report = {}
+    for repo in repositories:
+        repo_config = get_repo_information(client, repo, config.organization)
+        logger.debug(f"Repo configuration: {repo_config}")
+        actions = audit_actions(client, repo, config.organization)
+        logger.debug(f"Actions {actions}")
+        file_review = review_files(client, repo, config.organization)
+        logger.debug(f"Files: {file_review}")
+
+        checklist = actions | file_review | repo_config
+        auditing_rules = get_file()
+        score = get_result(checklist, auditing_rules["rules"])
+        logger.debug(f"Rules to follow: {auditing_rules}")
+
+        report[repo] = {
+            "score": score,
+            "result": "OK" if score > auditing_rules["threshold"] else "Action required",
+            "description": checklist
+        }
+
+    if config.save_results is True:
+        json_report = json.dumps(report, indent=4)
+
+        with open("reports/audit-result.json", "w") as outfile:
+            outfile.write(json_report)
+    else:
+        # Using print instead of logger to get result in JSON format
+        print(report)
+
+    logger.info(f"Finished auditing repositories for {config.organization} in {'{:.2f}'.format(time.time() - start)} seconds")
+
+
+def audit_actions(client: GitHubClient, repository: str, organization: str) -> dict:
+    audit_results = {
+        'Has Actions': False,
+        'Uses CodeQL': False,
+        'Uses Allowed list': False
+    }
+
+    actions = client.get_actions(organization, repository)
+
+    logger.debug(f"Got {actions['total_count']} workflow files")
+
+    audit_results['Has Actions'] = actions['total_count'] > 0
+
+    workflow_paths = [actions['workflows']['path'] for actions['workflows'] in actions['workflows']]
+    for file_path in workflow_paths:
+        file_content = client.get_file_content(organization, repository, file_path)
+        if not file_content:
+            logger.debug("File not found")
+            continue
+
+        if not audit_results['Uses CodeQL']:
+            audit_results['Uses CodeQL'] = "uses: github/codeql-action/analyze" in file_content
+
+        if not audit_results['Uses Allowed list']:
+            audit_results['Uses Allowed list'] = "uses: ed-fi-alliance-oss/ed-fi-actions/.github/workflows/repository-scanner.yml" in file_content
+
+    return audit_results
+
+
+def get_repo_information(client: GitHubClient, repository: str, organization: str) -> dict:
+    information = client.get_repository_information(organization, repository)
+
+    # Currently,this is only checking if there are alerts, which does not differentiates if dependabot is enabled or not
+    vulnerabilities = [alerts for alerts in information["vulnerabilityAlerts"]["nodes"]
+                       if (alerts["createdAt"] < (datetime.now() - timedelta(3 * 7)).isoformat() and
+                       alerts["securityVulnerability"]["advisory"]["severity"] in ['CRITICAL', 'HIGH'])]
```

Comment:
What do you think about extracting this array of two severities to a "constant" at the top of the file?